framework/src/meshgenerators/ExtraNodesetGenerator.C

```diff
@@ -132,13 +134,43 @@ ExtraNodesetGenerator::generate()
       this->comm().max(on_node);
     }
 
+    // only search for closest node if it is not found on a node
+    if (use_closest_node && !on_node)
+    {
+      // these are always true when using closest node
+      found_elem = true;
+      on_node = true;
+      Real dmin(std::numeric_limits<Real>::max());
+      dof_id_type dmin_id(std::numeric_limits<dof_id_type>::max());
+      // find this processors closest node and save distance and the node id
+      for (const auto & node_iter : as_range(mesh->local_nodes_begin(), mesh->local_nodes_end()))
+      {
+        Real dist = (*node_iter - p).norm_sq();
+        if (dist < dmin)
+        {
+          dmin = dist;
+          dmin_id = node_iter->id();
+        }
+      }
+
+      // get proc id with the global closest node
+      processor_id_type dmin_proc_id;
+      this->comm().minloc(dmin, dmin_proc_id);
+      // To make this work for distributed, only add on the proc_id containing the closest node
```

Comment:
No, this makes it wrong, both in the distributed case where `dmin_id` identifies a node that is ghosted on other ranks and in the *replicated* case (though it'll be easier there to avoid catching or tripping over the bug).

I'm going to be a little worried about my assertion coverage if this is passing in a `dbg` `DistributedMesh` sweep - in that configuration libMesh should be checking in `prepare_for_use()` that our boundary info is consistent across processors, and it shouldn't be *too* hard to trigger a case where this node is ghosted somewhere and the ghosted copy is inconsistent.

If you have a node id, use `mesh->query_node_ptr(dmin_id)` for each processor to see whether it owns that node; if you get `nullptr` back then you're done, and if you get a pointer back then `boundary_info.add_node()` with that pointer, so every processor that can see a node can see the correct boundary id on it.

Reply:
I should not have guessed

it fails -p 2

Reply:
It passes with -p 2, it fails with -p 2 and recover.

Reply:
@roystgnr How do I find the node id that is closest.  Should I use minloc to find the processor that had dmin and then use broadcast to send that processors node_id to every other processor and then query_node_ptr to see if that processor owns it.

Reply:
@roystgnr will you review this again.

Reply:
Current version looks like it's going to assertion fail in dbg, possibly segfault in opt.

I wasn't kidding about "use `mesh->query_node_ptr(dmin_id)`"  Unless you know that node `dmin_id` is local or ghosted on every processor, you cannot ask for `node_ptr(dmin_id)` because there is some processor on which no pointer that can be returned to you.

Reply:
@roystgnr I was checking that dmin_id was local before add_node:
```
      if (mesh->query_node_ptr(dmin_id))
        for (const auto & boundary_id : boundary_ids)
          boundary_info.add_node(mesh->node_ptr(dmin_id), boundary_id);
```

I changed it to this: ``boundary_info.add_node(mesh->query_node_ptr(dmin_id), boundary_id);``

Reply:
@roystgnr will you review the changes I made?

Reply:
Looks correct now, thanks.

I'd have cached the first `mesh->query_node_ptr(dmin_id)` (C++17 lets you put an init-statement inside the `if()` even), rather than calling it again for each `boundary_id` (where it's another O(log(Nnodes/Nprocs)) cost on a `DistributedMesh` for each call) ... but I'd have done that out of habit; performance is obviously not an issue here.

I just dismissed my previous review rather than adding an approval, though I can look at the rest of the changes more closely if Guillaume doesn't have time to re-review.

Reply:
I amended the commit with a cached node.  Thanks!

Reply:
Still looks good to me; thank you.  Even though caching isn't critical here, it's nice to see best practices so that others who read them are encouraged to do the same in places that do become hotspots.